src/AppBundle/Controller/APIController.php

```diff
@@ -160,6 +160,45 @@ public function registerAction(Request $request)
             ); 
         }
     }
+
+    /**
+     * @Route("/profile", name="profile")
+     */
+    public function profileAction(Request $request) 
+    {
+        $changePasswordModel = new ChangePassword();
+        $form = $this->createForm(ChangePasswordType::class, $changePasswordModel);
+
+        $form->handleRequest($request);
+        
+        $em = Utilities::getEntityManager($this);
+        $user = Utilities::getCurrentUser($this);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $password = $this->get('security.password_encoder')
+                ->encodePassword($user, $changePasswordModel->getNewPassword());
+            $user->setPassword($password);
+            $user->setUpdated(new \DateTime);
+            $em->persist($user);
+            $em->flush();
+            
+            return new JsonResponse(
+                array(
+                    'status' => 200, 
+                    'form' => $form->createView(),
+                    'flash' => "Successfully Updated Password!",
+                )
+            );
+        }
+
+        return new JsonResponse(
+                array(
+                    'status' => 200, 
+                    'form' => $form->createView(),
```

Comment:
This is not what we should be returning. Not supposed to be using a form. That is HTML rendering. Just need to return a serialized user and have the android team render out everything. If they want to change their password we need to make a separate form for that.